packages/react/src/lib/VoiceProvider.tsx

```diff
@@ -233,14 +233,20 @@ export const VoiceProvider: FC<VoiceProviderProps> = ({
 
   const mic = useMicrophone({
     streamRef,
-    onAudioCaptured: useCallback((arrayBuffer) => {
-      try {
-        client.sendAudio(arrayBuffer);
-      } catch (e) {
-        const message = e instanceof Error ? e.message : 'Unknown error';
-        updateError({ type: 'socket_error', message });
-      }
-    }, []),
+    onAudioCaptured: useCallback(
+      (arrayBuffer) => {
+        try {
+          if (client.readyState !== 'open') {
+            throw new Error('Socket is not open');
+          }
+          client.sendAudio(arrayBuffer);
+        } catch (e) {
+          const message = e instanceof Error ? e.message : 'Unknown error';
+          updateError({ type: 'socket_error', message });
+        }
+      },
```

Comment:
this try/catch isn't necessary if we have it on sendAudio

Reply:
I've udpated the functions, I think this should work now